app-misc/recoll/recoll-1.43.0.ebuild

```diff
@@ -0,0 +1,154 @@
+
+DESCRIPTION="Personal full text search package"
+HOMEPAGE="https://www.recoll.org"
+SRC_URI="https://www.recoll.org/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+
+IUSE="camelcase chm +inotify kde qt6 session +spell systemd webengine"
+REQUIRED_USE="
+	kde? ( qt6 )
+	session? ( inotify )
+	webengine? ( qt6 )
+	${PYTHON_REQUIRED_USE}
+"
+
+DEPEND="
+	dev-libs/libxml2
+	dev-libs/libxslt
+	dev-libs/xapian:=
+	sys-libs/zlib
+	virtual/libiconv
+	chm? (
+		dev-libs/chmlib
+		dev-python/pychm
+	)
+	kde? (
+		kde-plasma/plasma-workspace:6
+	)
+	qt6? (
+		dev-qt/qtbase:6[gui,network,widgets]
+		webengine? ( dev-qt/qtwebengine:6[widgets] )
+	)
+	session? (
+		inotify? (
+			x11-libs/libSM
+			x11-libs/libICE
+			x11-libs/libX11
+		)
+	)
+	spell? ( app-text/aspell )
+	systemd? ( sys-apps/systemd )
+	${PYTHON_DEPS}
+"
+
+BDEPEND="
+	qt6? ( dev-qt/qttools:6[linguist] )
+"
+
+RDEPEND="
+	${DEPEND}
+	app-arch/unzip
+"
+
+pkg_setup() {
+	if use inotify; then
+		local CONFIG_CHECK="~INOTIFY_USER"
+		check_extra_config
+	fi
+	python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+	python_fix_shebang filters
+}
+
+src_configure() {
+	use qt6 && export QMAKE="$(qt6_get_bindir)/qmake"
+
+	local emesonargs=(
+		$(meson_use camelcase)
+		$(meson_use chm python-chm)
+		$(meson_use inotify)
+		$(meson_use qt6 qtgui)
+		$(meson_use session x11mon)
+		$(meson_use spell aspell)
+		$(meson_use spell python-aspell)
+		$(meson_use systemd)
+		$(meson_use webengine)
+		-Dappimage=false
+		-Dfam=false
+		-Drecollq=true
+		-Dwebkit=false
+		-Dsystemd-system-unit-dir="$(systemd_get_systemunitdir)"
+		-Dsystemd-user-unit-dir="$(systemd_get_userunitdir)"
+	)
+
+	use qt6 && emesonargs+=( $(usex webengine "-Dwebpreview=true" "-Dwebpreview=false") )
+
+	meson_src_configure
+}
+
+src_install() {
+	meson_install
+	rm -rf "${D}/$(python_get_sitedir)"/*.egg-info || die
+	python_optimize
+
+	# html docs should be placed in /usr/share/doc/${PN}/html
+	rm -r "${ED}/usr/share/${PN}/doc" || die
+	find "${ED}" -name '*.la' -delete || die
+
+	# The recoll kio plugin and krunner plugins can be built
+	# and compiled only after recoll has been installed in $ED
+	if use kde; then
+		mkdir -p "${WORKDIR}/build_kio" "${WORKDIR}/build_krunner" || die
+		mv -f "${WORKDIR}/${P}/kde/kioslave/kio_recoll/"{CMakeLists-KF6.txt,CMakeLists.txt} || die
+		mv -f "${WORKDIR}/${P}/kde/krunner/"{CMakeLists-KF6.txt,CMakeLists.txt} || die
+
+		pushd "${WORKDIR}/build_kio" > /dev/null || die
+		cmake -S "${WORKDIR}/${P}/kde/kioslave/kio_recoll" \
+			-DCMAKE_INSTALL_PREFIX="${ED}/usr" \
+			-DDATA_INSTALL_DIR="${ED}/usr/share/recoll" || die
+		cmake --build . || die
+		cmake --install . || die
+		popd > /dev/null
+
+		pushd "${WORKDIR}/build_krunner" > /dev/null || die
+		cmake -S "${WORKDIR}/${P}/kde/krunner" -DCMAKE_INSTALL_PREFIX="${ED}/usr" || die
+		cmake --build . || die
+		cmake --install . || die
+		popd > /dev/null
+	fi
```

Comment:
No. That's a real abomination. As already said in https://github.com/gentoo/gentoo/pull/40099#issuecomment-2602918731 - separate packages.